view/frontend/web/js/recobuy.js

```diff
@@ -35,56 +35,78 @@
 
 define([
     'nostojs',
-    'jquery',
-    'catalogAddToCart'
+    'jquery'
 ], function (nostojs, $) {
     'use strict';
 
-    //noinspection SpellCheckingInspection
-    var form = $('#nosto_addtocart_form').catalogAddToCart({});
     var Recobuy = {};
+
     Recobuy.addProductToCart = function (productId, element, quantity) {
         quantity = quantity || 1;
         var productData = {
-            "productId" : productId,
-            'skuId' : productId
+            productId : productId,
+            skuId : productId,
+            quantity: quantity
         };
-        Recobuy.addSkuToCart(productData, element, quantity);
+        return Recobuy.addSkuToCart(productData, element);
     };
 
     // Products must be and array of objects [{'productId': '123', 'skuId': '321'}, {...}]
     // skuId is optional for simple products.
     Recobuy.addMultipleProductsToCart = function (products, element) {
         if (Array.isArray(products)) {
-            products.forEach(function (productObj) {
-                Recobuy.addSkuToCart(productObj, element, 1);
-            });
+            return Recobuy.recursiveAddProducts(products, element)
+        } else {
+            Promise.reject(new Error("Products is not type array"))
         }
     };
 
+    Recobuy.recursiveAddProducts = function (products, element) {
+        var length = products.length;
+        if (length === 0) {
+            return Promise.resolve();
+        }
+        return Recobuy.addSkuToCart(products[0], element, 1)
+            .then(function (res) {
+                return Recobuy.recursiveAddProducts(products.slice(1), element)
+            })
+    }
+
     // Product object must have fields productId and skuId {'productId': '123', 'skuId': '321'}
-    Recobuy.addSkuToCart = function (product, element, quantity) {
-        quantity = quantity || 1;
+    Recobuy.addSkuToCart = function (product, element) {
+
+        var quantity = product.quantity || 1;
```

Comment:
Does this work? https://magento.stackexchange.com/questions/131189/magento-2-what-is-the-use-of-formkey

Reply:
That's the way to add the form_key to the form. We use it here https://github.com/Nosto/nosto-magento2/blob/develop/view/frontend/templates/addtocart.phtml#L42